models/articlesModel.js

```diff
@@ -24,16 +22,24 @@ const fetchAllArticles = () => {
         votes,
         article_img_url,
         (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id) AS comment_count
-        FROM articles
-        ORDER BY created_at DESC`
-    )
-    .then((result) => {
-      // Remove body property from each article object
+        FROM articles`;
+
+  if (topic) {
+    query += ` WHERE topic = $1 ORDER BY created_at DESC`;
```

Comment:
You have used the if statement correctly if there's a topic but if you want to still always have an order by set then you can add this outside of the if statement afterwards. If you always add the ORDER BY line whether there's a topic or not, then it will always be there and in the correct place of SQL statements. Take it out of here and add to query after the block